qubesbuilder/plugins/installer/__init__.py

```diff
@@ -0,0 +1,515 @@
+            # Outside mock chroot to use the latest packages (dnf, openssl, etc.) from the cage.
+            #
+
+            cmd += [
+                f"sudo --preserve-env={','.join(self.environment.keys())} "
+                f"make -C {self.executor.get_plugins_dir()}/installer "
+                f"iso-prepare iso-packages-anaconda iso-packages-lorax",
+            ]
+
+            try:
+                self.executor.run(
+                    cmd,
+                    copy_in,
+                    copy_out,
+                    environment=self.environment,
+                    files_inside_executor_with_placeholders=files_inside_executor_with_placeholders,
+                )
+            except ExecutorError as e:
+                msg = f"{self.dist}: Failed to prepare installer: {str(e)}."
+                raise InstallerError(msg) from e
+
+        if stage == "build":
+            copy_in = [
+                (self.plugins_dir / "installer", self.executor.get_plugins_dir()),
+                (
+                    cache_dir / self.iso_name / "work",
+                    self.executor.get_plugins_dir() / "installer",
+                ),
+                (
+                    cache_dir / self.iso_name / "rpm",
+                    self.executor.get_plugins_dir() / "installer/yum/installer",
+                ),
+            ] + [
+                (self.plugins_dir / plugin, self.executor.get_plugins_dir())
+                for plugin in self.plugin_dependencies
+            ]
+
+            # Copy-in builder local repository
+            if repository_dir.exists():
+                copy_in += [(repository_dir, self.executor.get_repository_dir())]
+
+            # Add prepared chroot cache
+            if chroot_cache.exists():
+                copy_in += [(chroot_cache, Path("/var/cache/mock"))]
+
+            copy_out = [
+                (
+                    self.executor.get_plugins_dir()
+                    / f"installer/work/{self.iso_version}/{self.dist.architecture}/iso/{self.iso_name}.iso",
+                    iso_dir,
+                )
+            ]
+
+            # Prepare installer cmd
+            cmd = []
+
+            # Create builder-local repository (could be empty) inside the cage
+            cmd += [
+                f"mkdir -p {self.executor.get_repository_dir()}",
+                f"cd {self.executor.get_repository_dir()}",
+                "createrepo_c .",
+            ]
+
+            #
+            # Inside mock chroot to use qubes-lorax
+            #
+
+            mock_cmd = [
+                f"sudo --preserve-env={','.join(self.environment.keys())}",
+                f"/usr/libexec/mock/mock",
+                f"--root {self.executor.get_plugins_dir()}/installer/mock/{mock_conf}",
+                "--disablerepo='*'",
+                f"--chroot 'env {self.get_env()} make -C {self.executor.get_plugins_dir()}/installer iso-prepare iso-parse-kickstart iso-installer-lorax iso-installer-mkisofs'",
```

Comment:
iso-parse-kickstart not needed here, right?

Reply:
In fact I need it because it extracts also installer.repo which I give to lorax in args. Yes naming is not ideal.